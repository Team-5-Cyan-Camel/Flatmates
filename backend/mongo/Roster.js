const mongoose = require('mongoose');

const taskSchema = require('./Task');

const rosterSchema = mongoose.Schema({
    title: {
        type: String,
        required: true,
    },
    tasks: {
        type: [taskSchema]
    },
    assignedUsers: [{
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'User'
    }]
});

module.exports = mongoose.model('Roster', rosterSchema);
