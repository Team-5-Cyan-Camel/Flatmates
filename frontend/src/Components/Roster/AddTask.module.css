.modalContainer {
  top: 0;
  left: 0;
  width: 100vh !important;
  height: 100vh;
  overflow: hidden;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.3);
  z-index: 100;
}

.form {
  background-color: white;
  padding-top: 10px;
  padding-bottom: 20px;
  padding-left: 20px;
  padding-right: 20px;
}
