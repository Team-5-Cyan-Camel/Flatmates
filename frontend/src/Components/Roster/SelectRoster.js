import React from 'react';
import Card from 'react-bootstrap/Card';
import 'bootstrap/dist/css/bootstrap.min.css';
import Button from 'react-bootstrap/Button';
import '../Lobby/NavBar.css';
import AddRoster from './AddRoster';
import {useState, useEffect} from 'react';
import {FaTimes as Cross} from 'react-icons/fa';
import axios from 'axios';
import '../Lobby/NavBar.css';

import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import {Link, NavLink, useHistory, useParams} from 'react-router-dom';
import {confirmAlert} from 'react-confirm-alert'; // Import
import 'react-confirm-alert/src/react-confirm-alert.css'; // Import css

const SelectRoster = ({rosters, setRoster, updateDb, isHost}) => {
  let [makeRoster, setMakeRoster] = useState(false);

  const deleteRoster = (id) => {
    confirmAlert({
      title: 'Deleting Roster',
      message: 'Are you sure you want to delete this roster?',
      buttons: [
        {
          label: 'Ok',
          onClick: () => {
            const rostDel = {
              rosterID: id,
            };

            axios
              .delete('/roster', {data: rostDel})
              .then((res) => {})
              .catch(function (error) {
                console.log(error);
              });
          },
        },
        {
          label: 'Cancel',
        },
      ],
    });
  };

  return (
    <>
      <Navbar bg='dark' variant='dark' style={{}}>
        <Nav className='mr-auto'>
          {rosters !== null &&
            rosters.rosters.map((e, i) => [
              <Nav.Link
                key={i}
                onClick={() => setRoster(e.title)}
                style={{outline: '1px solid black ', outlineColor: 'black'}}
              >
                {e.title}{' '}
                {isHost && <Cross onClick={() => deleteRoster(e._id)} />}
              </Nav.Link>,
            ])}
        </Nav>
        <Button className='NavBarButton' onClick={() => setMakeRoster(true)}>
          add
        </Button>
      </Navbar>
      {makeRoster && <AddRoster show={setMakeRoster} updateDb={updateDb} />}
    </>
  );
};

export default SelectRoster;
