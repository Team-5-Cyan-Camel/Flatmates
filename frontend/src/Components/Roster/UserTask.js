import React from "react";
import Card from "react-bootstrap/Card";
import "bootstrap/dist/css/bootstrap.min.css";
import { useState, useEffect } from "react";
import AddTask from "./AddTask";
import { FaTimes as Cross } from "react-icons/fa";
import axios from "axios";

import { FaPlusCircle as PlusSymbol } from "react-icons/fa";

import Button from "react-bootstrap/Button";

const UserTask = ({ task, name, rid, pid, updateDb }) => {
  let [makeTask, setMakeTask] = useState(false);
  const deleteTask = (id) => {
    const delTask = {
      rosterID: rid,
      taskID: id,
    };

    axios
      .delete("/roster/task", { data: delTask })
      .then((res) => {})
      .catch(function (error) {
        console.log(error);
      });
  };

  return (
    <>
      <Card
        id="Card-field"
        className="UsersTasks"
        style={{
          alignItems: "center",

          width: "20em",
          maxHeight: "100%",
          margin: "1em",
          height: "100%",
        }}
      >
        <Card.Header
          as="h5"
          id="Card-Header"
          className="text-center"
          style={{ width: "100%" }}
        >
          {" "}
          {name}
        </Card.Header>

        <Card.Body
          style={{
            borderLeft: "#454549 solid 1px",
            borderRight: "#454549 solid 1px",
            display: "flex",
            flexDirection: "column",

            alignItems: "flex-start",

            height: "100%",
            width: "100%",
            overflowY: "auto",
            backgroundColor: "#7e828b",
          }}
        >
          {task.map((e) => {
            return (
              <Card
                id="Card-field"
                style={{
                  alignItems: "center",
                  justifyContent: "center",
                  outline: "black solid 1px",
                  margin: "0.25em",
                  borderRadius: "0",
                  width: "100%",
                }}
              >
                <Card.Body
                  style={{
                    alignItems: "center",

                    width: "100%",
                    outlineColor: "black",

                    display: "flex",
                    overflowX: "hidden",
                  }}
                >
                  <p>
                    <Cross onClick={() => deleteTask(e._id)} />
                    {e.title} {": "}{" "}
                    {e.description === "" ? "N/A" : e.description}
                  </p>
                </Card.Body>
              </Card>
            );
          })}
        </Card.Body>
        <Card.Footer id="Card-Footer">
          <Button
            className="CardButton text-center"
            onClick={() => setMakeTask(true)}
            style={{ verticalAlign: "middle" }}
          >
            <PlusSymbol style={{ verticalAlign: "middle" }}></PlusSymbol>
          </Button>
        </Card.Footer>
      </Card>

      {makeTask && (
        <AddTask show={setMakeTask} rid={rid} pid={pid} updateDb={updateDb} />
      )}
    </>
  );
};

export default UserTask;
