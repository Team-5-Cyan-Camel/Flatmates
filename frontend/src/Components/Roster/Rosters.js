import Roster from "./Roster";
import React, { useState, useEffect } from "react";
import SelectRoster from "./SelectRoster";
import axios from "axios";

const Rosters = ({ room, isHost }) => {
  let [Room, setRoom] = useState(room);
  let [DisplayRoster, setDisplayRoster] = useState(null);

  useEffect(() => {
    setRoom(room);
    // get roster data
    if (Room !== null && Room.rosters.length !== 0) {
      setDisplayRoster({ rosters: Room.rosters[0] });
    }
  }, [room]);

  useEffect(() => {
    axios
      .get("/room")
      .then((res) => {
        setRoom(res.data);
      })
      .catch(function (error) {
        console.log(error);
      });
  }, []);

  const roster = (title) => {
    for (var roster of Room.rosters) {
      if (roster.title === title) {
        setDisplayRoster({ rosters: roster });
        break;
      }
    }
  };

  return (
    <div style={{ width: "85vw" }}>
      <SelectRoster isHost={isHost} room={Room} setRoster={roster} />

      {Room !== null && DisplayRoster !== null && (
        <Roster data={DisplayRoster.rosters} isHost={isHost} />
      )}
    </div>
  );
};

export default Rosters;
