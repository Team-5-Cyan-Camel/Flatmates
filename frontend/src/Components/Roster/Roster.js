import React, { useState } from "react";
import Card from "react-bootstrap/Card";
import "bootstrap/dist/css/bootstrap.min.css";
import Button from "react-bootstrap/Button";
import UserTask from "./UserTask";
import "../../App.css";
import axios from "axios";

const Roster = ({ data, updateDb, isHost }) => {
  let [ifHost, setIsHost] = useState(isHost);

  const rotate = () => {
    const rotateRep = {
      rosterID: data._id,
    };

    axios
      .patch("/roster/rotate", rotateRep)
      .then((res) => {})
      .catch(function (error) {
        console.log(error);
      });
  };
  return (
    <>
      {data !== "undefined" && (
        <Card
          id="Card-field"
          style={{
            alignItems: "center",
            justifyContent: "center",
            height: "60vh",
            maxHeight: "70vh",
            borderTopRightRadius: "0",
            borderTopLeftRadius: "0",
          }}
        >
          <Card.Header
            as="h5"
            id="Card-Header"
            className="text-center"
            style={{
              width: "100%",
              borderTopRightRadius: "0",
              borderTopLeftRadius: "0",
              alignItems: "center",
              justifyContent: "center",
              display: "grid",
              gridTemplateColumns: "1fr 7fr 1fr",
            }}
          >
            <div></div> asdasd
            {data.title}
            {ifHost && (
              <Button
                className="GoButton"
                style={{ margin: "0" }}
                onClick={rotate}
              >
                rotate
              </Button>
            )}
          </Card.Header>

          <Card.Body
            style={{
              display: "Flex",
              alignItems: "flex-start",
              justifyContent: "center",
              width: "90%",
              maxHeight: "100%",
            }}
          >
            {data.assignedUsers.map((e, i) => {
              return (
                <UserTask
                  key={i}
                  rid={data._id}
                  pid={e._id}
                  name={e.name}
                  updateDb={updateDb}
                  task={data.tasks.filter((data) => {
                    return data.userIndex === i;
                  })}
                />
              );
            })}
          </Card.Body>
        </Card>
      )}
    </>
  );
};

export default Roster;
