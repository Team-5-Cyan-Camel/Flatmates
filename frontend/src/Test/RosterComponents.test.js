import React from "react";
import renderer from "react-test-renderer";
import Roster from "./../Components/Roster/Roster";
import Rosters from "./../Components/Roster/Rosters";
import SelectRoster from "./../Components/Roster/SelectRoster";
import UserTask from "./../Components/Roster/UserTask";

const room = {
  users: [
    {
      _id: "6094f73787fe4e0ff89ef600",
      username: "UK",
      email: "",
      name: "Leon",
      phoneNumber: "",
    },
    {
      _id: "6094ff16dba18c4bcc3e376e",
      username: "coke",
      email: "",
      name: "Coco",
      phoneNumber: "",
    },
    {
      _id: "608674f890c6b43a80d04f8e",
      username: "loine",
      name: "Leon",
      phoneNumber: "021 0817 9478",
      email: "whomst@gmail.com",
    },
  ],
  _id: "6094f73d87fe4e0ff89ef601",
  host: "6094f73787fe4e0ff89ef600",
  rosters: [
    {
      assignedUsers: [
        {
          _id: "6094f73787fe4e0ff89ef600",
          username: "UK",
          name: "Leon",
        },
        {
          _id: "6094ff16dba18c4bcc3e376e",
          username: "coke",
          name: "Coco",
        },
        {
          _id: "608674f890c6b43a80d04f8e",
          username: "loine",
          name: "Leon",
        },
      ],
      _id: "6094fb79cd7f9941445dd4a1",
      title: "nice",
      tasks: [],
    },
    {
      assignedUsers: [
        {
          _id: "6094f73787fe4e0ff89ef600",
          username: "UK",
          name: "Leon",
        },
        {
          _id: "6094ff16dba18c4bcc3e376e",
          username: "coke",
          name: "Coco",
        },
        {
          _id: "608674f890c6b43a80d04f8e",
          username: "loine",
          name: "Leon",
        },
      ],
      _id: "6096785e44457b3fbca1f9b2",
      title: "nice",
      tasks: [],
    },
    {
      assignedUsers: [
        {
          _id: "6094f73787fe4e0ff89ef600",
          username: "UK",
          name: "Leon",
        },
        {
          _id: "6094ff16dba18c4bcc3e376e",
          username: "coke",
          name: "Coco",
        },
        {
          _id: "608674f890c6b43a80d04f8e",
          username: "loine",
          name: "Leon",
        },
      ],
      _id: "6096787044457b3fbca1f9b4",
      title: "mine",
      tasks: [],
    },
  ],
  reminders: [],
  expenses: [],
  __v: 16,
};

test("Roster renders correctly", () => {
  const tree = renderer

    .create(<Roster data={room.rosters[0]} isHost={true} updateDb={null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Rosters container (containing tabs and roster view) render correctly", () => {
  const tree = renderer

    .create(<Rosters rosters={room} isHost={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Roster tabs render correctly", () => {
  const tree = renderer

    .create(
      <SelectRoster
        isHost={true}
        rosters={room}
        setRoster={null}
        updateDb={null}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("User task list in roster render correctly", () => {
  const roster = room.rosters[0];
  const userTasks = roster.assignedUsers[0];
  const tree = renderer

    .create(
      <UserTask
        key={0}
        rid={userTasks._id}
        pid={userTasks._id}
        name={userTasks.name}
        updateDb={null}
        task={roster.tasks.filter((data) => {
          return data.userIndex === 0;
        })}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
