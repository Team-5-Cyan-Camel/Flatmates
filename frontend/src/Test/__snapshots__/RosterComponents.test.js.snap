// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster renders correctly 1`] = `
<div
  className="card"
  id="Card-field"
  style={
    Object {
      "alignItems": "center",
      "borderTopLeftRadius": "0",
      "borderTopRightRadius": "0",
      "height": "65vh",
      "justifyContent": "center",
      "maxHeight": "80vh",
      "width": "100%",
    }
  }
>
  <h5
    className="text-center card-header"
    id="Card-Header"
    style={
      Object {
        "alignItems": "center",
        "borderTopLeftRadius": "0",
        "borderTopRightRadius": "0",
        "display": "grid",
        "gridTemplateColumns": "1fr 7fr 1fr",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <div />
    nice
    <button
      className="NavBarButton btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
        />
      </svg>
    </button>
  </h5>
  <div
    className="UsersTasksList card-body"
    style={
      Object {
        "alignItems": "center",
        "display": "grid",
        "gridAutoFlow": "column",
        "minWidth": "0",
        "overflowX": "auto",
        "padding": "1em",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "grid",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <div
        className="UsersTasks card"
        id="Card-field"
        style={
          Object {
            "alignItems": "center",
            "height": "47vh",
            "margin": "0.5em",
            "maxHeight": "100%",
            "width": "19em",
          }
        }
      >
        <h5
          className="text-center card-header"
          id="Card-Header"
          style={
            Object {
              "width": "100%",
            }
          }
        >
           
          Leon
        </h5>
        <div
          className="card-body"
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#7e828b",
              "borderLeft": "#454549 solid 1px",
              "borderRight": "#454549 solid 1px",
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "overflowY": "auto",
              "width": "100%",
            }
          }
        />
        <div
          className="card-footer"
          id="Card-Footer"
        >
          <button
            className="CardButton text-center btn btn-primary"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "verticalAlign": "middle",
              }
            }
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "grid",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <div
        className="UsersTasks card"
        id="Card-field"
        style={
          Object {
            "alignItems": "center",
            "height": "47vh",
            "margin": "0.5em",
            "maxHeight": "100%",
            "width": "19em",
          }
        }
      >
        <h5
          className="text-center card-header"
          id="Card-Header"
          style={
            Object {
              "width": "100%",
            }
          }
        >
           
          Coco
        </h5>
        <div
          className="card-body"
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#7e828b",
              "borderLeft": "#454549 solid 1px",
              "borderRight": "#454549 solid 1px",
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "overflowY": "auto",
              "width": "100%",
            }
          }
        />
        <div
          className="card-footer"
          id="Card-Footer"
        >
          <button
            className="CardButton text-center btn btn-primary"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "verticalAlign": "middle",
              }
            }
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "grid",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <div
        className="UsersTasks card"
        id="Card-field"
        style={
          Object {
            "alignItems": "center",
            "height": "47vh",
            "margin": "0.5em",
            "maxHeight": "100%",
            "width": "19em",
          }
        }
      >
        <h5
          className="text-center card-header"
          id="Card-Header"
          style={
            Object {
              "width": "100%",
            }
          }
        >
           
          Leon
        </h5>
        <div
          className="card-body"
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#7e828b",
              "borderLeft": "#454549 solid 1px",
              "borderRight": "#454549 solid 1px",
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "overflowY": "auto",
              "width": "100%",
            }
          }
        />
        <div
          className="card-footer"
          id="Card-Footer"
        >
          <button
            className="CardButton text-center btn btn-primary"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "verticalAlign": "middle",
              }
            }
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer"
    id="Card-Footer"
  />
</div>
`;

exports[`Roster tabs render correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
  style={Object {}}
>
  <div
    className="mr-auto navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "marginLeft": "0.25em",
          "marginRight": "0.25em",
        }
      }
    >
      nice
       
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "marginLeft": "0.25em",
          "marginRight": "0.25em",
        }
      }
    >
      nice
       
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "marginLeft": "0.25em",
          "marginRight": "0.25em",
        }
      }
    >
      mine
       
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </a>
  </div>
  <button
    className="NavBarButton btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
      />
    </svg>
  </button>
</nav>
`;

exports[`Rosters container (containing tabs and roster view) render correctly 1`] = `
<div
  style={
    Object {
      "width": "85vw",
    }
  }
>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
    style={Object {}}
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "marginLeft": "0.25em",
            "marginRight": "0.25em",
          }
        }
      >
        nice
         
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "marginLeft": "0.25em",
            "marginRight": "0.25em",
          }
        }
      >
        nice
         
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "marginLeft": "0.25em",
            "marginRight": "0.25em",
          }
        }
      >
        mine
         
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </a>
    </div>
    <button
      className="NavBarButton btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
        />
      </svg>
    </button>
  </nav>
</div>
`;

exports[`User task list in roster render correctly 1`] = `
<div
  className="UsersTasks card"
  id="Card-field"
  style={
    Object {
      "alignItems": "center",
      "height": "47vh",
      "margin": "0.5em",
      "maxHeight": "100%",
      "width": "19em",
    }
  }
>
  <h5
    className="text-center card-header"
    id="Card-Header"
    style={
      Object {
        "width": "100%",
      }
    }
  >
     
    Leon
  </h5>
  <div
    className="card-body"
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#7e828b",
        "borderLeft": "#454549 solid 1px",
        "borderRight": "#454549 solid 1px",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "overflowY": "auto",
        "width": "100%",
      }
    }
  />
  <div
    className="card-footer"
    id="Card-Footer"
  >
    <button
      className="CardButton text-center btn btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "verticalAlign": "middle",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
        />
      </svg>
    </button>
  </div>
</div>
`;
